seahorn/lib/array_list_helper.c

```diff
@@ -19,6 +19,21 @@ void initialize_array_list(struct aws_array_list *const list) {
   list->data = can_fail_malloc(list->current_size);
   list->alloc = sea_allocator();
 }
+
+void initialize_bounded_array_list(struct aws_array_list *const list,
+                                const size_t max_initial_size,
+                                const size_t max_initial_item_allocation,
+                                const size_t max_item_size) {
+  list->current_size = nd_size_t();
```

Comment:
should call `assume(aws_array_list_is_bounded(...))` to avoid replicating the conditions all over the place

The function itself is a little fishy because it does not relate `item_size * length` to `current_size`
Maybe a call to `assume(aws_array_list_is_valid(...))` is also appropriate to be placed within this function.